onnxruntime/contrib_ops/cpu/bert/multihead_attention.cc

```diff
@@ -137,6 +143,39 @@ Status MultiHeadAttention<T>::Compute(OpKernelContext* context) const {
   ORT_RETURN_IF_ERROR(MaybeTransposeToBNSHAndAddBias<T>(
       context, allocator, batch_size, num_heads_, kv_sequence_length, v_head_size, value, bias, v_bias_offset, V));
 
+  if (std::is_same_v<T, float> && !disable_flash_ && key_padding_mask == nullptr && extra_add_qk == nullptr && past_key == nullptr && past_value == nullptr) {
```

Comment:
Also check is_unidirectional_ flag. I think current implementation does not support that.

Reply:
Added checks. Thanks.